### PR TITLE
Fixed copying of filter and PID objects

### PR DESCRIPTION
**Pull request by tridge**
This fixes a valgrind error with AP_Soaring in SITL, caused by the initialised variable in LowPassFilter from (ironically) not being initialised when a filter is copied to init.
To make this easier to apply in more places I've added a CLASS_NO_COPY() macro which can be used to forbid copies in the class definition
If we like this pattern then we can use it in a lot more places

### CODE REVIEW

File: libraries/AP_Soaring/Variometer.cpp [hunk 1]
```diff
@@ -9,9 +9,8 @@ Manages the estimation of aircraft total energy, drag and vertical air velocity.
 Variometer::Variometer(const AP_Vehicle::FixedWing &parms) :
     _aparm(parms)
 {
-    _climb_filter = LowPassFilter<float>(1.0/60.0);
-
-    _vdot_filter2 = LowPassFilter<float>(1.0f/60.0f);
+    _climb_filter.set_cutoff_frequency(1.0/60);
+    _vdot_filter2.set_cutoff_frequency(1.0/60);
```
Comment (peterbarker) [inline libraries/AP_Soaring/Variometer.cpp:13]: Aren't these redundant given the in-header initialisation?

(well, for `_climb_filter` - shouldn't the init of `_vdo_filter` be the same, 'though?

**Review by peterbarker — COMMENTED** [review]
I like this.

**Comment by rmackay9** [comment]
I'm happy with the wheel encoder and PID changes..

File: libraries/AP_Soaring/Variometer.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_Soaring/Variometer.cpp:13]: fixed

**Comment by peterbarker** [comment]
Failed with an internal error, probably same one @joshanne reported

```
2021-06-07T02:13:09.5112327Z AT-0425.8: Exception caught: Did not get SYSTEM_TIME message after 60.000000 seconds
2021-06-07T02:13:09.5113554Z Traceback (most recent call last):
2021-06-07T02:13:09.5114661Z   File "/__w/ardupilot/ardupilot/Tools/autotest/common.py", line 5782, in run_one_test_attempt
2021-06-07T02:13:09.5118565Z     test_function()
2021-06-07T02:13:09.5119463Z   File "/__w/ardupilot/ardupilot/Tools/autotest/arducopter.py", line 6648, in fly_each_frame
2021-06-07T02:13:09.5120340Z     self.do_RTL()
2021-06-07T02:13:09.5121180Z   File "/__w/ardupilot/ardupilot/Tools/autotest/arducopter.py", line 435, in do_RTL
2021-06-07T02:13:09.5122050Z     self.change_mode("RTL")
2021-06-07T02:13:09.5122938Z   File "/__w/ardupilot/ardupilot/Tools/autotest/common.py", line 4469, in change_mode
2021-06-07T02:13:09.5123818Z     tstart = self.get_sim_time()
2021-06-07T02:13:09.5124707Z   File "/__w/ardupilot/ardupilot/Tools/autotest/common.py", line 2691, in get_sim_time
2021-06-07T02:13:09.5126039Z     raise AutoTestTimeoutException("Did not get SYSTEM_TIME message after %f seconds" % timeout)
2021-06-07T02:13:09.5127636Z common.AutoTestTimeoutException: Did not get SYSTEM_TIME message after 60.000000 seconds
2021-06-07T02:13:09.5128534Z 
2021-06-07T02:13:09.5131890Z AT-0425.8: log list: ['logs/00000001.BIN', 'logs/00000002.BIN', 'logs/00000003.BIN', 'logs/00000004.BIN', 'logs/00000005.BIN', 'logs/00000006.BIN', 'logs/00000007.BIN', 'logs/00000008.BIN', 'logs/00000009.BIN', 'logs/00000010.BIN', 'logs/00000011.BIN', 'logs/00000012.BIN', 'logs/00000013.BIN', 'logs/00000014.BIN', 'logs/00000015.BIN', 'logs/00000016.BIN', 'logs/00000017.BIN', 'logs/00000018.BIN', 'logs/00000019.BIN', 'logs/00000020.BIN', 'logs/00000021.BIN', 'logs/00000022.BIN', 'logs/00000023.BIN', 'logs/00000024.BIN', 'logs/00000025.BIN', 'logs/00000026.BIN', 'logs/00000027.BIN', 'logs/00000028.BIN', 'logs/00000029.BIN', 'logs/00000030.BIN', 'logs/00000031.BIN', 'logs/00000032.BIN', 'logs/00000033.BIN', 'logs/00000034.BIN', 'logs/00000035.BIN', 'logs/00000036.BIN', 'logs/00000037.BIN', 'logs/00000038.BIN', 'logs/00000039.BIN', 'logs/00000040.BIN']
2021-06-07T02:13:09.5133719Z AT-0425.8: Most recent logfile: logs/00000040.BIN
2021-06-07T02:13:09.5134406Z AT-0425.8: set_parameters: ({})
2021-06-07T02:13:09.5135121Z AT-0425.8: Drained 0 messages from mav (0.000000/s)
2021-06-07T02:13:09.5136338Z SCurve::calculate_path invalid outputs
2021-06-07T02:13:09.5136908Z PANIC: internal error 536870912
2021-06-07T02:13:09.5137655Z Running: sh Tools/scripts/dumpstack.sh 3496 >dumpstack_arducopter.3496.out 2>&1
2021-06-07T02:13:09.5138324Z Stack dumped
2021-06-07T02:13:09.5138712Z [New LWP 3497]
2021-06-07T02:13:09.5139244Z [Thread debugging using libthread_db enabled]
2021-06-07T02:13:09.5140184Z Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
2021-06-07T02:13:09.5141129Z 0x00007f9141079dff in wait4 () from /lib/x86_64-linux-gnu/libc.so.6
2021-06-07T02:13:09.5142074Z #0  0x00007f9141079dff in wait4 () from /lib/x86_64-linux-gnu/libc.so.6
2021-06-07T02:13:09.5142595Z No symbol table info available.
2021-06-07T02:13:09.5143288Z #1  0x00007f9140fe90e7 in ?? () from /lib/x86_64-linux-gnu/libc.so.6
2021-06-07T02:13:09.5143825Z No symbol table info available.
2021-06-07T02:13:09.5144463Z #2  0x000055a5d54a4702 in AP_HAL::dump_stack_trace() ()
2021-06-07T02:13:09.5144963Z No symbol table info available.
2021-06-07T02:13:09.5145458Z #3  0x000055a5d54a499f in AP_HAL::panic(char const*, ...) ()
2021-06-07T02:13:09.5145964Z No symbol table info available.
2021-06-07T02:13:09.5146649Z #4  0x000055a5d53e06eb in AP_InternalError::error(AP_InternalError::error_t, unsigned short) ()
2021-06-07T02:13:09.5147540Z No symbol table info available.
2021-06-07T02:13:09.5148363Z #5  0x000055a5d551a9ed in SCurve::calculate_path(float, float, float, float, float, float, float&, float&, float&, float&) const ()
2021-06-07T02:13:09.5149176Z No symbol table info available.
2021-06-07T02:13:09.5149819Z #6  0x000055a5d551b4ee in SCurve::add_segments(float) [clone .part.0] ()
2021-06-07T02:13:09.5150460Z No symbol table info available.
2021-06-07T02:13:09.5151340Z #7  0x000055a5d551c417 in SCurve::calculate_track(Vector3<float> const&, Vector3<float> const&, float, float, float, float, float, float, float) ()
2021-06-07T02:13:09.5152282Z No symbol table info available.
2021-06-07T02:13:09.5152973Z #8  0x000055a5d54603a5 in AC_WPNav::set_wp_destination(Vector3<float> const&, bool) ()
2021-06-07T02:13:09.5153762Z No symbol table info available.
2021-06-07T02:13:09.5154363Z #9  0x000055a5d5460db0 in AC_WPNav_OA::set_wp_destination(Vector3<float> const&, bool) ()
2021-06-07T02:13:09.5154969Z No symbol table info available.
2021-06-07T02:13:09.5155794Z #10 0x000055a5d545effb in AC_WPNav::set_wp_destination_loc(Location const&) ()
2021-06-07T02:13:09.5156648Z No symbol table info available.
2021-06-07T02:13:09.5157249Z #11 0x000055a5d5340454 in ModeRTL::climb_start() ()
2021-06-07T02:13:09.5157836Z No symbol table info available.
2021-06-07T02:13:09.5158384Z #12 0x000055a5d534154d in ModeRTL::run() ()
2021-06-07T02:13:09.5158937Z No symbol table info available.
2021-06-07T02:13:09.5159516Z #13 0x000055a5d5327142 in Copter::fast_loop() ()
2021-06-07T02:13:09.5160090Z No symbol table info available.
2021-06-07T02:13:09.5160711Z #14 0x000055a5d53db4de in AP_Scheduler::loop() ()
2021-06-07T02:13:09.5161302Z No symbol table info available.
2021-06-07T02:13:09.5161898Z #15 0x000055a5d53dfd84 in AP_Vehicle::loop() ()
2021-06-07T02:13:09.5162483Z No symbol table info available.
2021-06-07T02:13:09.5163193Z #16 0x000055a5d549adb6 in HAL_SITL::run(int, char* const*, AP_HAL::HAL::Callbacks*) const ()
2021-06-07T02:13:09.5163905Z No symbol table info available.
```

@khancyr no symbol table available?  Is it possible we're not compiling with debug symbols now?!

**Review by peterbarker — APPROVED** [review]
LGTM.

Even saves 80 bytes on Plane

**Pull request merged**